themes/developers/lib/views/primary-nav-menu.php

```diff
@@ -10,8 +10,8 @@
  */
 
 namespace spiralWebDB\Developers;
-?>
-<nav class="nav-primary" aria-label="Main" itemscope itemtype="https://schema.org/SiteNavigationElement" id="genesis-nav-primary">
+
+return '<nav class="nav-primary" aria-label="Main" itemscope itemtype="https://schema.org/SiteNavigationElement" id="genesis-nav-primary">
```

Comment:
@rgadon107 Why `return` here. This is a view file.

Reply:
I went ahead and approved it. But you may need to fix this ^.

Reply:
@hellofromtonya The `return` control structure is absolutely needed in order for the filtered `$navoutput` to work on the front end. It needs to remain in the filter function. 

However, I came up with an alternative solution to a calling a view file. I brought the filtered string literal back into the function scope, returned it, and and wrapped the string using PHP's heredoc syntax. It works, and eliminates the need for an additional view directory and file in the child theme. You demonstrated the use of heredoc tags on 2 separate strings in PR #32.

Reply:
@hellofromtonya The `donate` class attribute is added to the nav menu list item. It’s added as a class to the first `<li>` tag in the menu. 

I went into the admin at Appearance >> Menu and inspected the ‘Donate’ menu item (the one I’m targeting with the `donate` class attribute). I could add a `<span>` tag around the term ‘Donate’, in order to pass a class attribute. But that would not allow me to target the background color around the nav menu item. The `<span>` would only target the menu term itself. 

Earlier you wrote: 

>How is this menu item added? Is it through the Appearance > Menu interface in the admin? _If yes, you can add class attributes to any item in that interface, i.e. without code._

It’s not evident to me how a class attribute can be added to a nav menu list item. Can you demonstrate how that would done? 

I’m disappointed that the `genesis_do_nav` filter won’t work. The filter doesn’t allow one to drill down into individual menu list items to add a class. The only way to do it is to return _everything_ inside the `<nav></nav>` tags as a string, and then modify the string as needed. 

![add donate class attribute to nav menu item PR 52](https://user-images.githubusercontent.com/3718603/72551050-a635a600-3859-11ea-8086-6ce49da88ca7.png)

Reply:
@hellofromtonya Commit  https://github.com/rgadon107/cornerstone/pull/52/commits/7644cf59474a7e66437862ec578532589b3e906a removes the file  at `/developers/lib/functions/menu.php`, and adds a`donate` class attribute to the primary nav menu via a CSS Classes field on the site admin at Appearance >> Menus.